### ISSUE TITLE
Floyd-Warshall implementation doesn't work on undirected graphs.

### ISSUE THREAD

**Issue by cryslith** [petgraph/petgraph#524]
```
use petgraph::{graph::{UnGraph, NodeIndex}, algo::floyd_warshall::floyd_warshall}; // 0.6.2

fn main() {
    let mut g: UnGraph<(), ()> = UnGraph::new_undirected();
    let n1 = g.add_node(());
    let n2 = g.add_node(());
    g.add_edge(n1, n2, ());
    println!("{:?}", floyd_warshall(&g, |_| 1));
}
```

outputs `Ok({(NodeIndex(0), NodeIndex(0)): 0, (NodeIndex(1), NodeIndex(0)): 2147483647, (NodeIndex(1), NodeIndex(1)): 0, (NodeIndex(0), NodeIndex(1)): 1})`

I expect the distance from node 1 to node 0 to be 1 rather than the max integer.

**Comment by Dr-Emann** [comment]
This appears to have been fixed in #487, but hasn't been released yet.

**Comment by jfarrell468** [comment]
I encountered this as well.

**Comment by ABorgna** [comment]
The bugfix has been published in 0.6.3.
